src/core/internal/hash.d

```diff
@@ -130,16 +130,10 @@ private template canBitwiseHash(T)
 
 //enum hash. CTFE depends on base type
 size_t hashOf(T)(auto ref T val, size_t seed = 0)
-if (is(T EType == enum) && (!__traits(isScalar, T) || is(T == __vector)))
```

Comment:
The original code didn't behave as intended because `is(T == __vector)` is always false if `is(T == enum)`.